src/server/cluster/cluster_config.cc

```diff
@@ -25,29 +27,95 @@ std::string_view ClusterConfig::KeyTag(std::string_view key) {
   return key.substr(start + 1, end - start - 1);
 }
 
-SlotId ClusterConfig::KeySlot(std::string_view key) {
-  std::string_view tag = KeyTag(key);
+SlotId ClusterConfig::KeySlot(string_view key) {
+  string_view tag = KeyTag(key);
   return crc16(tag.data(), tag.length()) & kMaxSlotNum;
 }
 
-ClusterConfig::ClusterConfig() {
+ClusterConfig::ClusterConfig(string_view my_id) : my_id_(my_id) {
   cluster_enabled = true;
-  AddSlots();
 }
 
-void ClusterConfig::AddSlots() {
-  // TODO update logic acording to config
-  // currently add all slots to owned slots
-  std::lock_guard lk{slots_mu_};
-  for (SlotId slot_id = 0; slot_id <= kMaxSlotNum; ++slot_id) {
-    owned_slots_.emplace(slot_id);
+bool ClusterConfig::IsConfigValid(const vector<ClusterShard>& new_config) {
+  // Make sure that all slots are set exactly once.
+  array<bool, tuple_size<decltype(slots_)>::value> slots_found = {};
+  for (const auto& shard : new_config) {
+    for (const auto& slot_range : shard.slot_ranges) {
+      if (slot_range.start > slot_range.end) {
+        return false;
```

Comment:
Please add log prints so it will be easier to understand if we have a problem with the config

Reply:
Oh, that's a great idea!